### PR TITLE
Raster management

### PR DESCRIPTION
**Pull request by anikaweinmann**
Basic Raster management added:
- /locations/{location_name}/mapsets/{mapset_name}/raster_layers - GET
- /locations/{location_name}/mapsets/{mapset_name}/raster_layers/{raster_name} - GET, DELETE, POST

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,3 +11,8 @@ test:
 	pip install -r $(SRC)/requirements.txt
 	pip install pytest
 	python -m pytest tests/
+
+devtest:
+	pip install -r $(SRC)/requirements.txt
+	pip install pytest
+	python -m pytest tests/ -m dev
```
Comment (marcjansen) [inline Makefile:18]: This line errors oin my machine with sth along the lines:

```
python -m pytest -m dev tests/ 
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/mjansen/code/actinia-python-client
collected 14 items / 14 deselected / 0 selected                                

============================ 14 deselected in 0.13s ============================
make: *** [Makefile:18: devtest] Fehler 5
```

File: actinia/raster.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#######
+# acintia-python-client is a python client for actinia - an open source REST
```
Comment (marcjansen) [inline actinia/raster.py:4]:
```suggestion
# actinia-python-client is a python client for actinia - an open source REST
```

File: actinia/raster.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#
+#######
+
+__license__ = "GPLv3"
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
+__maintainer__ = "Anika Weinmann"
+
+import json
+# import requests
```
Comment (marcjansen) [inline actinia/raster.py:31]:
```suggestion
```

File: actinia/raster.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+from actinia.region import Region
+from actinia.utils import request_and_check, print_stdout
```
Comment (marcjansen) [inline actinia/raster.py:34]:
```suggestion
from actinia.utils import request_and_check
```

File: actinia/raster.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                tbres=None,
+                rows=r_info["rows"],
+                cols=r_info["cols"],
+                rows3=None,
+                cols3=None,
+                depths=None,
+                cells=r_info["cells"],
+                cells3=None
+            )
+        print_stdout(json.dumps(self.info, indent=4))
```
Comment (marcjansen) [inline actinia/raster.py:79]:
```suggestion
```

File: actinia/utils.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#######
+# acintia-python-client is a python client for actinia - an open source REST
```
Comment (marcjansen) [inline actinia/utils.py:4]:
```suggestion
# actinia-python-client is a python client for actinia - an open source REST
```

File: actinia/utils.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def request_and_check(url, auth, status_code=200):
+    resp = requests.get(url, auth=auth)
+    if resp.status_code != status_code:
+        raise Exception(f"Error {resp.status_code}: {resp.text}")
+    return json.loads(resp.text)
+
+
+def print_stdout(msg):
```
Comment (marcjansen) [inline actinia/utils.py:42]: A follow-up PR might introduce a real logging library, that can be turned on and off

File: actinia/utils.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
+__maintainer__ = "Anika Weinmann"
+
+import json
+import requests
+import sys
+
+
+def request_and_check(url, auth, status_code=200):
```
Comment (marcjansen) [inline actinia/utils.py:35]: API docs are missing

File: tests/test_actinia_raster_management.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#######
+# acintia-python-client is a python client for actinia - an open source REST
```
Comment (marcjansen) [inline tests/test_actinia_raster_management.py:4]:
```suggestion
# actinia-python-client is a python client for actinia - an open source REST
```

File: tests/test_actinia_raster_management.py [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+    get_raster_info_mock,
+    raster_info_resp,
+    upload_raster_resp,
+    delete_raster_resp,
+    start_job_resp,
+)
+
+__license__ = "GPLv3"
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
```
Comment (marcjansen) [inline tests/test_actinia_raster_management.py:54]:
```suggestion
```

File: tests/mock/actinia_raster_management_mock.py [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+
+__license__ = "GPLv3"
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
+__maintainer__ = "Anika Weinmann"
+
+from .actinia_mock import (
+    ACTINIA_BASEURL,
+    ACTINIA_VERSION,
+    # fill_basic_frame,
```
Comment (marcjansen) [inline tests/mock/actinia_raster_management_mock.py:33]:
```suggestion
    ACTINIA_VERSION
```

**Review by marcjansen — APPROVED** [review]
`make test` passes for me, while `make devtest` doesn't. Overall, this looks good to me.

File: Makefile (hunk 1 above)
Comment (anikaweinmann) [reply Makefile:18]: I will add a description how you can use it

File: actinia/utils.py (hunk 8 above)
Comment (anikaweinmann) [reply actinia/utils.py:35]: Will be added

File: Makefile (hunk 1 above)
Comment (anikaweinmann) [reply Makefile:18]: @marcjansen please check if it is now ok

Comment (marcjansen) [reply Makefile:18]: Great, now I understand that this expected behaviour. Thanks for the clarification

File: actinia/raster.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        self.name = name
+        self.__location_name = location_name
+        self.__mapset_name = mapset_name
+        self.__actinia = actinia
+        self.__auth = auth
+        self.region = None
+        self.info = None
+
+    def get_info(self):
+        """Return the informations of the raster map
```
Comment (marcjansen) [inline actinia/raster.py:45]:
```suggestion
        """Return the information of the raster map
```

File: actinia/utils.py [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+    Parameters:
+        url (string): URL as string
+        auth (tuple): Tupel of user and password
+        status_code (int): Status code to check if it is set; default is 200
+
+    Returns:
+        (dict): returns text of the response as dictionary
+
+    Throw s an error if the request does not have the status_code
```
Comment (marcjansen) [inline actinia/utils.py:46]:
```suggestion
    Throws an error if the request does not have the status_code
```

**Pull request merged**